File: clients/filesystem-hadoop3/src/main/java/com/datastrato/gravitino/filesystem/hadoop/context/FileSystemContext.java
/*
 * Copyright 2024 Datastrato Pvt Ltd.
 * This software is licensed under the Apache License version 2.
 */

package com.datastrato.gravitino.filesystem.hadoop.context;

import com.datastrato.gravitino.secret.Secret;
import org.apache.hadoop.fs.FileSystem;

/** Context for FileSystem. */
public interface FileSystemContext {

  /**
   * Get the FileSystem.
   *
   * @return FileSystem
   */
  FileSystem getFileSystem();

  /**
   * Get the Secret.
   *
   * @return Secret
   */
  Secret getSecret();

  /** Close the context */
  void close();
}

File: clients/client-python/gravitino/filesystem/filesystem_context.py
"""
Copyright 2024 Datastrato Pvt Ltd.
This software is licensed under the Apache License version 2.
"""

from abc import ABC, abstractmethod

from fsspec import AbstractFileSystem


class FileSystemContext(ABC):
    """Context for FileSystem."""

    @abstractmethod
    def get_filesystem(self) -> AbstractFileSystem:
        pass

    @abstractmethod
    def get_secret(self) -> str:
        pass
